### ISSUE TITLE
can this be compiled for ARM based platform?

### ISSUE THREAD

**Issue by kulpatel** [clicon/cligen#87]
can this be compiled for ARM based platform?

**Comment by rcmcdonald91** [comment]
Yes. As an example, the FreeBSD port builds on both armv7 and aarch64.

**Comment by kulpatel** [comment]
@rcmcdonald91 thanks for your reply.
As of now i m using below command to copmile it for x86.
```
cd cligen
./configure
make all 
```
May i know, do i need to change it in make file or just change in command with diff argument will do the job? if so can you share the sample command please.

**Comment by kulpatel** [comment]
@rcmcdonald91 @shmuelnatan @mgsmith1000 @pheller @Tagrenam @nowaits @andrnils @loos-br @dcornejo @StasSt-siklu @krihal @dcornejo @baruchsiach @olofhagsand @rbgarga  can someone help me with above query. 

```
As of now i m using below command to copmile it for x86.

cd cligen
./configure
make all 
May i know, do i need to change it in make file or just change in command with diff argument will do the job? if so can you share the sample command please
```

**Comment by olofhagsand** [comment]
configure uses config.guess to guess the architecture of the machine you are running on. Ie, if you are on a native ARM arch and run configure it will most probably guess you are on ARM and produce binaries for ARM. But this is useful only if your build arch and target arch are the same.
But if you are on a n x86 but want to produce ARM binaries you need to cross-compile and the commands above are not appropriate. 
Many that cross-compile cligen/clixon use buildroot / yocto for that and there are some discussions here and there about it,, but not summarized on one page as far as I know. You can also see what was done for openwrt (a variant of buildroot) here: https://github.com/clicon/clixon-openwrt.
Otherwise you should be able to search how to cross-compile using autoconf for generic guidelines.
There is also some activity in switching to CMake which (they say?) provides better support for cross-compiling.
Hope that answers your question.
